### DISCUSSION TITLE
Timeline duration with repeat

### DISCUSSION THREAD

**Discussion by Coffee2JS** [motiondivision/motionone#153] [Support]
Hello!
I'm just getting started with MotionOne, and ran into a problem with timeline duration: docs say it gets calculated, but I don't seem to get it working. 

For example, I added 2 keyframess to timeline, repeating in 3 times, expecting duration to be 12:

` const tl = timeline([
['#img', {rotate: 180}, {duration: 1 }],
            ['#img', {rotate: 0}, {duration: 3 }],
        ], {repeat: 3});
console.log('duration:', tl.duration); // this is always 4, regradless of repeat value
`

Adding repeats to individual KFs had no effect either (though I'm not sure if that's supported) . What am I doing wrong here?

**Comment by mattgperry** [comment]
The `repeat` is saying the animation itself will be repeated 3 times. The animation is 4 seconds long and repeated 3 times.

**Reply by Coffee2JS** [reply]
OK, and thanks for the quick reply.

Then I need to calculate it myself... I was expecting something similar to GSAP's totalDuration(). Does timeline duration include delays etc.? And is it possible to repeat keyframes or parts of sequence on a timeline? If so, are those calculated into the duration?

**Reply by mattgperry** [reply]
It includes `delay` within the timeline keyframes themselves but again not on the overall animation (so not `delay` passed via the second argument)

Not currently possible to repeat keyframes, I am looking into this in Framer Motion so could backport it, and yes they would be calculated into the duration (however it probably wouldn't be possible to do `repeat: Infinity` for keyframes for this reason)
